src/app/pages/storage/modules/pool-manager/components/configuration-preview/configuration-preview.component.ts

```diff
@@ -20,7 +21,14 @@ export class ConfigurationPreviewComponent {
 
   protected name$ = this.store.name$;
   protected encryption$ = this.store.encryption$;
-  protected topology$ = this.store.topology$;
+  protected topology$ = this.store.usesDraidLayout$.pipe(switchMap((usesDraidLayout) => {
```

Comment:
This will limit `topology$` to only emit when `usesDraidLayout$` emits first. At least we should be using the `map` operator on `topoology$`